packages/webpack-module-federation/src/loadRemote.ts

```diff
@@ -1,7 +1,17 @@
-import { isNil } from "@squide/core";
+import { type ModuleRegisterFunction, isNil } from "@squide/core";
+
+// The interface of a module that can be loaded dynamically.
+interface Module {
+    register?: ModuleRegisterFunction;
```

Comment:
I am not sure about this. At some point something else could be loaded than our standard register function. It's up to the consumer.

Reply:
The idea is we own the code that allows us to register remote modules into the host. in the `registerRemoteModules` we already look for `module.register`. The client can register whatever they want, be we still need to manipulate those registration here, so it make sense to define what we are expecting. 

the `registerRemoteModules` file isnt shown in this review because it hasn't changed, but it uses what has been registered, so i really think this type is necessary